corehq/pillows/app_submission_tracker.py

```diff
@@ -126,3 +131,89 @@ def get_sql_app_form_submission_tracker_reindexer():
         FormReindexAccessor(include_attachments=False)
     )
     return AppFormSubmissionReindexer(doc_provider, FORM_SQL, 'form_processor_xforminstancesql')
+
+
+@quickcache(['domain'], timeout=60 * 60 * 5)
+def _get_apps_for_domain(domain):
+    project = Domain.get_by_name(domain)
+    if project:
+        return [app._id for app in project.applications()]
+    else:
+        return []
+
+
+class UserAppFormSubmissionDocProcessor(BaseDocProcessor):
+    def __init__(self, pillow_processor):
+        self.pillow_processor = pillow_processor
+
+    def process_doc(self, doc):
+        form_submission_changes = self._doc_to_changes(doc)
+        for change in form_submission_changes:
+            try:
+                self.pillow_processor.process_change(None, change)
+            except Exception:
+                return False
+        return True
+
+    def handle_skip(self, doc):
+        print('Unable to process user {}'.format(
+            doc['_id'],
+        ))
+        return True
+
+    def _doc_to_changes(self, doc):
```

Comment:
This could do with a comment to say why it's converting a user doc into xform changes